experimental/segmenter/src/language.rs

```diff
@@ -52,74 +47,61 @@ pub fn get_language(codepoint: u32) -> Language {
 /// Actually supported LSTM model is Thai and Burmese only. If using other
 /// code point, it causes panic.
 pub struct LanguageIterator<'s> {
-    input: Chars<'s>,
-    current_ch: Option<char>,
+    rest: &'s str,
 }
 
 impl<'s> LanguageIterator<'s> {
     #[allow(dead_code)]
     pub fn new(input: &'s str) -> Self {
-        let mut input = input.chars();
-        let current_ch = input.next();
-        Self { input, current_ch }
+        Self { rest: input }
     }
 }
 
 impl<'s> Iterator for LanguageIterator<'s> {
-    type Item = String;
+    type Item = &'s str;
 
     fn next(&mut self) -> Option<Self::Item> {
-        let mut s = String::new();
-
-        let current_ch = self.current_ch?;
-        let lang = get_language(current_ch as u32);
-        s.push(current_ch);
-        for c in self.input.by_ref() {
-            self.current_ch = Some(c);
-            let new_lang = get_language(c as u32);
-            if lang != new_lang {
-                return Some(s);
+        let mut indices = self.rest.char_indices();
+        let lang = get_language(indices.next()?.1 as u32);
+        match indices.find(|&(_, ch)| get_language(ch as u32) != lang) {
+            Some((i, _)) => {
+                let (result, rest) = self.rest.split_at(i);
```

Comment:
Note: `split_at` is panicky but this call site is fine; there is no clippy lint for the panicky slice methods other than indexing_slicing